ext/dom/php_dom.stub.php

```diff
@@ -170,67 +170,67 @@ class DOMAttr extends DOMNode
 
-    /** @return string */
-    public function getAttributeNS(?string $namespace, string $localName) {}
+    /** @tentative-return-type */
+    public function getAttributeNS(?string $namespace, string $localName): string {}
 
-    /** @return DOMAttr|DOMNamespaceNode|false */
-    public function getAttributeNode(string $qualifiedName) {}
+    /** @tentative-return-type */
+    public function getAttributeNode(string $qualifiedName): DOMAttr|DOMNameSpaceNode|false {}
 
-    /** @return DOMAttr|DOMNamespaceNode|null */
-    public function getAttributeNodeNS(?string $namespace, string $localName) {}
+    /** @tentative-return-type */
+    public function getAttributeNodeNS(?string $namespace, string $localName): DOMAttr|DOMNameSpaceNode|null {}
 
-    /** @return DOMNodeList */
-    public function getElementsByTagName(string $qualifiedName) {}
+    /** @tentative-return-type */
+    public function getElementsByTagName(string $qualifiedName): DOMNodeList {}
 
-    /** @return DOMNodeList */
-    public function getElementsByTagNameNS(?string $namespace, string $localName) {}
+    /** @tentative-return-type */
+    public function getElementsByTagNameNS(?string $namespace, string $localName): DOMNodeList {}
 
-    /** @return bool */
-    public function hasAttribute(string $qualifiedName) {}
+    /** @tentative-return-type */
+    public function hasAttribute(string $qualifiedName): bool {}
 
-    /** @return bool */
-    public function hasAttributeNS(?string $namespace, string $localName) {}
+    /** @tentative-return-type */
+    public function hasAttributeNS(?string $namespace, string $localName): bool {}
 
-    /** @return bool */
-    public function removeAttribute(string $qualifiedName) {}
+    /** @tentative-return-type */
+    public function removeAttribute(string $qualifiedName): bool {}
 
-    /** @return void */
-    public function removeAttributeNS(?string $namespace, string $localName) {}
+    /** @tentative-return-type */
+    public function removeAttributeNS(?string $namespace, string $localName): void {}
 
-    /** @return DOMAttr|false */
-    public function removeAttributeNode(DOMAttr $attr) {}
+    /** @tentative-return-type */
+    public function removeAttributeNode(DOMAttr $attr): DOMAttr|false {}
 
-    /** @return DOMAttr|bool */
-    public function setAttribute(string $qualifiedName, string $value) {}
+    /** @tentative-return-type */
+    public function setAttribute(string $qualifiedName, string $value): DOMAttr|bool {}
 
-    /** @return void */
-    public function setAttributeNS(?string $namespace, string $qualifiedName, string $value) {}
+    /** @tentative-return-type */
+    public function setAttributeNS(?string $namespace, string $qualifiedName, string $value): void {}
 
-    /** @return DOMAttr|null|false */
-    public function setAttributeNode(DOMAttr $attr) {}
+    /** @tentative-return-type */
+    public function setAttributeNode(DOMAttr $attr): DOMAttr|null|false {}
```

Comment:
Something that bothers me about the DOM case is that this puts a number of `false` return values into the contract that are only relevant without strict errors. Such as the one here. Once the return types are in place, dropping `|false` will be breaking change :/

Reply:
It looks like `strictErrorChecking` has actually been removed from the DOM spec, so I have some hope that we can simply drop support for it. Needs some investigation to understand what the impact is and maybe looking at some of the current uses (https://beta.grep.app/search?q=-%3EstrictErrorChecking%5Cs%2A%3D%5Cs%2Afalse&regexp=true).